criu/arch/aarch64/crtools.c

```diff
@@ -20,10 +21,87 @@
 #include "cpu.h"
 #include "restorer.h"
 #include "compel/infect.h"
+#include "pstree.h"
+
+extern unsigned long getauxval(unsigned long type);
 
 #define assign_reg(dst, src, e) dst->e = (__typeof__(dst->e))(src)->e
 
-int save_task_regs(void *x, user_regs_struct_t *regs, user_fpregs_struct_t *fpsimd)
+static int save_pac_keys(int pid, CoreEntry *core)
+{
+	struct user_pac_address_keys paca;
+	struct user_pac_generic_keys pacg;
+	PacKeys *pac_entry;
+	long pac_enabled_key;
+	struct iovec iov;
+	int ret;
+
+	unsigned long hwcaps = getauxval(AT_HWCAP);
+
+	hwcaps = 0;
+
+	pr_debug("%d: Dumping PAC keys\n", pid);
+	pac_entry = xmalloc(sizeof(PacKeys));
+	if (!pac_entry)
+		return -1;
+	core->ti_aarch64->pac_keys = pac_entry;
+	pac_keys__init(pac_entry);
+
+	if (hwcaps & HWCAP_PACA) {
```

Comment:
Would it be possible to add a comment here that `HWCAP_PACG` indicates whether PAC is supported for pointers?

Perhaps we should move the `pr_debug("%d: Dumping PAC keys\n", pid);` inside this if statement?